### DISCUSSION TITLE
How to use Android StorageManager.PrimaryStorageVolume.CreateOpenDocumentTreeIntent?

### DISCUSSION THREAD

**Discussion by r2d2Proton** [dotnet/maui#17200] [Q&A]
I am trying to display an Android System dialog to install my app. Currently I have the following:

`
var service = MauiApplication.Current.GetSystemService(Context.StorageService);
Android.OS.Storage.StorageManager sm = service as Android.OS.Storage.StorageManager;
var primVol = sm.PrimaryStorageVolume;
var intent = primVol.CreateOpenDocumentTreeIntent();
intent.SetFlags(ActivityFlags.NewTask);

string startDir = "Documents";
Java.Lang.Object? result = intent.GetParcelableExtra("android.provider.extra.INITIAL_URI");
MauiApplication.Current.StartActivity(intent);
`

This shows the attached dialog at startup. I don't know how to wait for it to finish.

![CreateOpenDocumentTreeIntent](https://github.com/dotnet/maui/assets/4054178/843fcc0f-c1b1-4686-b0c2-66f99b04e13b)

**Comment by MatthiasGvonOptris** [comment]
https://stackoverflow.com/questions/67552027/android-11-action-open-document-tree-set-initial-uri-to-the-documents-folder/67554693#67554693
